semgrep/semgrep/error.py

```diff
@@ -88,7 +88,9 @@ class LegacySpan:
     config_path: Tuple[str]
 
 
-@attr.s(auto_attribs=True, frozen=True)
+# Don't use frozen=True because it's violated and raises an exception
+# while we're trying to create a clean SemgrepCoreError exception.
+@attr.s(auto_attribs=True)
 class SemgrepCoreError(SemgrepError):
     code: int
     level: Level
```

Comment:
^ The real fix is this.

Reply:
??? What is going on here? What is this frozen thing in the first place?

Reply:
Frozen is to make it hashable. Do you have a repro of this causing an infinite loop?